src/config.ts

```diff
@@ -1,22 +1,35 @@
+import CartesiDAppFactoryArbitrum from '@cartesi/rollups/deployments/arbitrum/CartesiDAppFactory.json';
+import InputBoxArbitrum from '@cartesi/rollups/deployments/arbitrum/InputBox.json';
+import CartesiDAppFactoryArbitrumGoerli from '@cartesi/rollups/deployments/arbitrum_goerli/CartesiDAppFactory.json';
+import InputBoxArbitrumGoerli from '@cartesi/rollups/deployments/arbitrum_goerli/InputBox.json';
 import CartesiDAppFactoryMainnet from '@cartesi/rollups/deployments/mainnet/CartesiDAppFactory.json';
 import InputBoxMainnet from '@cartesi/rollups/deployments/mainnet/InputBox.json';
 import CartesiDAppFactorySepolia from '@cartesi/rollups/deployments/sepolia/CartesiDAppFactory.json';
 import InputBoxSepolia from '@cartesi/rollups/deployments/sepolia/InputBox.json';
-import mainnet from '@cartesi/rollups/export/abi/mainnet.json';
+import rollupsMainnet from '@cartesi/rollups/export/abi/mainnet.json';
 import { lookupArchive } from '@subsquid/archive-registry';
-import { DataSource } from '@subsquid/evm-processor';
+import { GatewaySettings, RpcEndpointSettings } from '@subsquid/evm-processor';
+import { arbitrum, arbitrumGoerli, mainnet, sepolia } from 'viem/chains';
 
 // addresses are the same on all chains
 export const CartesiDAppFactoryAddress =
-    mainnet.contracts.CartesiDAppFactory.address.toLowerCase();
+    rollupsMainnet.contracts.CartesiDAppFactory.address.toLowerCase();
 export const ERC20PortalAddress =
-    mainnet.contracts.ERC20Portal.address.toLowerCase();
-export const InputBoxAddress = mainnet.contracts.InputBox.address.toLowerCase();
+    rollupsMainnet.contracts.ERC20Portal.address.toLowerCase();
+export const InputBoxAddress =
+    rollupsMainnet.contracts.InputBox.address.toLowerCase();
 export const ERC721PortalAddress =
-    mainnet.contracts.ERC721Portal.address.toLowerCase();
+    rollupsMainnet.contracts.ERC721Portal.address.toLowerCase();
+export const AuthorityFactoryAddress =
+    '0x519421Bd7843e0D1E2F280490962850e31c86087'.toLowerCase();
```

Comment:
You need to update the `@cartesi/rollups` package to version `1.2.0` and use the new `AuthorityFactory` they have.

The `AuthorityFactory` in the sunodo codebase was delete in the latest codebase.